src/librustc/mir/mod.rs

```diff
@@ -162,7 +160,7 @@ pub struct Body<'tcx> {
     pub span: Span,
 
     /// A cache for various calculations.
-    cache: cache::Cache,
+    predecessors_cache: Option<IndexVec<BasicBlock, Vec<BasicBlock>>>,
```

Comment:
Do we have any `skip_serializing` and `serde(default)` like attributes? If not, maybe we do need a Cache type which is just a newtype around `T: Default` and implements serialize and deserialize to do nothing. Otherwise we may regress memory usage and disk usage a lot

Reply:
I asked around, but the response was that there isn't any of that available. I switched back to the `Cache` struct.